src/com/lilithsthrone/game/character/CharacterUtils.java

```diff
@@ -1399,16 +1409,18 @@ public static void equipClothing(GameCharacter character, boolean replaceUnsuita
 							AbstractClothingType ct = getClothingTypeForSlot(character, slot, clothingToUse);
 							
 							clothingToUse.remove(ClothingType.PENIS_CONDOM);
+							AbstractClothing clothingToAdd = AbstractClothingType.generateClothing(
+									ct,
+									(slot == InventorySlot.GROIN || slot==InventorySlot.CHEST || slot==InventorySlot.SOCK
+											? ct.getAvailablePrimaryColours().contains(lingerieColour)?lingerieColour:ct.getAvailablePrimaryColours().get(Util.random.nextInt(ct.getAvailablePrimaryColours().size()))
+											: (slot.isCoreClothing()
+													?ct.getAvailablePrimaryColours().contains(primaryColour)?primaryColour:ct.getAvailablePrimaryColours().get(Util.random.nextInt(ct.getAvailablePrimaryColours().size()))
+													:ct.getAvailablePrimaryColours().contains(secondaryColour)?secondaryColour:ct.getAvailablePrimaryColours().get(Util.random.nextInt(ct.getAvailablePrimaryColours().size())))),
+									false);
+							clothingToAdd.setPattern(clothingPattern);
 							
 							if(ct!=null) {
-								character.equipClothingFromNowhere(AbstractClothingType.generateClothing(
```

Comment:
are you sure it's okay that the null check is done later? I'm not entirely sure myself but I'm just scrolling through the code